dbgap_extract.py

```diff
@@ -5,15 +5,29 @@
 import sys
 from datetime import datetime
 import logging
+import os
 
 FILENAME = "extract-" + datetime.now().strftime("%m-%d-%Y-%H-%M-%S")
 REQUEST_URL = "https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetSampleStatus.cgi?study_id={}&rettype=xml"
-
-logging.basicConfig(filename=FILENAME + ".log", level=logging.DEBUG)
-logging.getLogger().addHandler(logging.StreamHandler(sys.stdout))
-logging.getLogger("requests").setLevel(logging.WARNING)
-logging.getLogger("urllib3").setLevel(logging.WARNING)
-
+LOG_FILE = FILENAME + ".log"
+logging.basicConfig(filename=LOG_FILE, level=logging.DEBUG)
+
+def setup_logging(log_filename):
+    global LOG_FILE
+    if os.path.exists(LOG_FILE):
+        os.remove(LOG_FILE)
+    LOG_FILE = log_filename
+    fileh = logging.FileHandler(LOG_FILE, "a")
+    log = logging.getLogger()
+    for hdlr in log.handlers[:]:
+        log.removeHandler(hdlr)
+    log.setLevel(logging.DEBUG)
+    log.addHandler(fileh)
+    log.addHandler(logging.StreamHandler(sys.stdout))
+    logging.getLogger("requests").setLevel(logging.WARNING)
+    logging.getLogger("urllib3").setLevel(logging.WARNING)
+
+setup_logging(LOG_FILE)
```

Comment:
actually, we could probably avoid calling this from the global level and call it when you know the name for the log file (either inputed name or datetime)